File: feerrott/Agent_of_Change.lua
local fearplane_raid = {
    expedition = { name="Plane of Fear", min_players=1, max_players=40},
    instance   = { zone="fearplane", version=0, duration=eq.seconds("8h") }, -- zone lasts 8 hours
    safereturn = { zone="fearplane", x=2491, y=-2159.26, z=4, h=227.0 }, --outside portal
    zonein     = { x=1032, y=-834, z=100, h=260 },
}

function event_say(e)
	local dz = e.other:GetExpedition()
	if(e.message:findi("hail")) then
		if dz.valid and dz:GetZoneName() == "fearplane" then
			e.self:Say("Tell me when you're [" .. eq.say_link("ready") .. "] to enter")
		else
			e.self:Say("Would you like to [" .. eq.say_link("request") .. "] the expedition?")
		end
	elseif(e.message:findi("request")) then
		local dz = e.other:CreateExpedition(fearplane_raid)
		dz:AddReplayLockout(eq.seconds("9h")) -- 9 hour lockout
		e.self:Say("Tell me when you're [" .. eq.say_link("ready") .. "] to enter")
	elseif(dz:GetZoneName() == "fearplane" and e.message:findi("ready")) then
		e.other:MovePCDynamicZone("fearplane")
	end
end

File: feerrott/Agent_of_Change.pl
#agent of change by osirous

my $expedition_name = "Plane of Fear";
my $min_players     = 1;
my $max_players     = 40;
my $dz_zone         = "fearplane";
my $dz_version      = 0;
my $dz_duration     = 28800; # 8 hours | #3600; # 1 hour

sub EVENT_SAY {
  if ($text =~ /hail/i) {
    my $dz = $client->GetExpedition();
    if ($dz && $dz->GetName() eq $expedition_name) {
      quest::say("Tell me when you're [" . quest::saylink("ready") . "] to enter");
    }
    else {
      quest::say("Would you like to [" . quest::saylink("request") . "] the expedition?");
    }
  }
  elsif ($text =~ /request/i) {
    my $dz = $client->CreateExpedition($dz_zone, $dz_version, $dz_duration, $expedition_name, $min_players, $max_players);
    if ($dz) {
      $dz->SetSafeReturn("feerrott", 2491, -2159.26, 4, 227.0); # outside portal
      $dz->SetZoneInLocation(1032, -834, 100, 260); # normal zone in place
      $dz->AddReplayLockout(32400); # immediately add a 9 hour replay lockout on creation
      quest::say("Tell me when you're [" . quest::saylink("ready") . "] to enter");
    }
  }
  elsif ($text =~ /ready/i) {
    $client->MovePCDynamicZone("fearplane");
  }
}